### PR TITLE
allow using system libs

### PR DESCRIPTION
**Pull request by alucryd**
This makes it possible to use system libpng, minizip, libxxhash and zlib. Builds and work fine on Arch Linux.

### CODE REVIEW

**Comment by m4xw** [comment]
This will blow up with #195 
Rebase on the "rebase" branch and also please target the PR to "rebase".
It still has a few days to go, otherwise this needs to wait till the release got into "develop"

File: Makefile.common [hunk 1]
```diff
@@ -133,6 +129,24 @@ SOURCES_C = \
 	$(PNG_DIR)/pngwrite.c \
 	$(PNG_DIR)/pngwtran.c \
 	$(PNG_DIR)/pngwutil.c
```
Comment (m4xw) [inline Makefile.common:131]: \n

File: Makefile.common [hunk 2]
```diff
@@ -133,6 +129,24 @@ SOURCES_C = \
...
+XXHASH_SOURCES_C = \
+	$(XXHASH_DIR)/xxhash.c
```
Comment (m4xw) [inline Makefile.common:133]: same line, remove \

File: Makefile.common [hunk 3]
```diff
@@ -102,22 +107,13 @@ SOURCES_C = \
...
-	$(ZLIB_DIR)/inftrees.c \
-	$(ZLIB_DIR)/trees.c \
-	$(ZLIB_DIR)/uncompr.c \
-	$(ZLIB_DIR)/zutil.c \
+	$(CORE_DIR)/src/plugin/dummy_input.c
+
+MINIZIP_SOURCES_C = \
+	$(MINIZIP_DIR)/zip.c \
+	$(MINIZIP_DIR)/unzip.c \
+	$(MINIZIP_DIR)/ioapi.c
```
Comment (m4xw) [inline Makefile.common:115]: \n

**Comment by alucryd** [comment]
All done!

**Comment by m4xw** [comment]
Whats the reason behind this PR btw?
A missmatch between the versions can cause unintended side effects, there have been issues in the past with distribution's libpng being too old and causing crashes after a while and the likes.
If I merge this PR, be aware, I will only accept Issues with the embedded libs.

File: Makefile.common [hunk 4]
```diff
@@ -142,12 +158,41 @@ SOURCES_C += \
 	$(CORE_DIR)/subprojects/md5/md5.c
 COREFLAGS += -I$(CORE_DIR)/subprojects/md5
 
+# libpng
+ifeq ($(SYSTEM_LIBPNG), 1)
```
Comment (m4xw) [inline Makefile.common:162]: The last change i'd like before merge is adding each SYSTEM_ var to `Makefile` at the top, seperately spaced by `\n` with `?= 0`

File: Makefile.common [hunk 5]
```diff
@@ -142,12 +158,41 @@ SOURCES_C += \
...
+	SOURCES_C += $(LIBPNG_SOURCES_C)
+endif
+
 # Minizip
-SOURCES_C += \
-	$(CORE_DIR)/subprojects/minizip/zip.c \
-	$(CORE_DIR)/subprojects/minizip/unzip.c \
-	$(CORE_DIR)/subprojects/minizip/ioapi.c
-COREFLAGS += -I$(CORE_DIR)/subprojects/minizip
+ifeq ($(SYSTEM_MINIZIP), 1)
```
Comment (m4xw) [inline Makefile.common:171]: see comment

File: Makefile.common (hunk 4 above)
Comment (m4xw) [reply Makefile.common:162]: Purely so i keep track of them :P

Comment (alucryd) [reply Makefile.common:167]: Sure thing!

File: Makefile.common [hunk 6]
```diff
@@ -1,14 +1,22 @@
...
 
+MINIZIP_DIR        = $(CORE_DIR)/subprojects/minizip
+PNG_DIR            = $(ROOT_DIR)/custom/dependencies/libpng
+XXHASH_DIR         = $(CORE_DIR)/subprojects/xxhash
+ZLIB_DIR           = $(ROOT_DIR)/custom/dependencies/libzlib
+
+SYSTEM_MINIZIP ?= 0
+SYSTEM_LIBPNG ?= 0
+SYSTEM_XXHASH ?= 0
+SYSTEM_ZLIB ?= 0
```
Comment (m4xw) [inline Makefile.common:18]: I meant in `Makefile`, not in `Makefile.common`.
`Makefile.common` is not supposed to have any configuration vars like this, DYNAREC_USED is only a temp var and a exception.
After thats its good2go.

Comment (alucryd) [reply Makefile.common]: Ah , sorry, should be good now.

**Comment by m4xw** [comment]
@alucryd You think it would make sense to introduce non-pkg-config flags and have it use as fallback?
I could see this being used in some static platforms and there pkg-config wont be available.

**Comment by m4xw** [comment]
Actually I guess for those it should already resolve handle it in the frontend and it would need to be a "dont include dependency" flag, so guess out of scope.

**Comment by dmrlawson** [comment]
There are some inconsistencies with variable naming. I managed to get it to compile with these changes but I don't have time to test it right now:

```
$ git diff
diff --git a/Makefile.common b/Makefile.common
index f939d95..a212b34 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -9,7 +9,7 @@ VIDEODIR_ANGRYLION = $(ROOT_DIR)/mupen64plus-video-angrylion
 
 MINIZIP_DIR        = $(CORE_DIR)/subprojects/minizip
 PNG_DIR            = $(ROOT_DIR)/custom/dependencies/libpng
-XXHASH_DIR         = $(CORE_DIR)/subprojects/xxhash
+XXHASH_DIR         = $(ROOT_DIR)/xxHash
 ZLIB_DIR           = $(ROOT_DIR)/custom/dependencies/libzlib
 
 DYNAREC_USED := 0
@@ -29,7 +29,7 @@ INCFLAGS += \
        -I$(VIDEODIR_GLIDEN64)/src/inc
 
 MINIZIP_INCFLAGS = -I$(MINIZIP_DIR)
-PNG_INCFLAGS     = -I$(PNG_DIR)
+LIBPNG_INCFLAGS  = -I$(PNG_DIR)
 XXHASH_INCFLAGS  = -I$(XXHASH_DIR)
 ZLIB_INCFLAGS    = -I$(ZLIB_DIR)
 
@@ -114,7 +114,7 @@ MINIZIP_SOURCES_C = \
        $(MINIZIP_DIR)/unzip.c \
        $(MINIZIP_DIR)/ioapi.c
 
-PNG_SOURCES_C = \
+LIBPNG_SOURCES_C = \
        $(PNG_DIR)/png.c \
        $(PNG_DIR)/pngerror.c \
        $(PNG_DIR)/pngget.c \
@@ -131,8 +131,6 @@ PNG_SOURCES_C = \
        $(PNG_DIR)/pngwtran.c \
        $(PNG_DIR)/pngwutil.c
 
-XXHASH_SOURCES_C = $(XXHASH_DIR)/xxhash.c
-
 ZLIB_SOURCES_C = \
        $(ZLIB_DIR)/adler32.c \
        $(ZLIB_DIR)/compress.c \
@@ -182,7 +180,6 @@ ifeq ($(SYSTEM_XXHASH), 1)
        LDFLAGS += $(shell pkg-config --libs libxxhash)
 else
        INCFLAGS += $(XXHASH_INCFLAGS)
-       SOURCES_C += $(XXHASH_SOURCES_C)
 endif
 
 # zlib
```

**Comment by m4xw** [comment]
@alucryd See dmrlawson's diff, I didn't spot it, can you push a fix?

**Comment by m4xw** [comment]
Merged in c8af7dd8a031fd37e01d5be78909a1de9d80768b

**Comment by alucryd** [comment]
Apologies, I didn't get a chance to go over this in the past few days. Thanks for merging!

**Pull request closed without merging**